File: algds/codewars/4kyu/all-balanced-parenthesis/rb/parens.spec.rb
require 'rspec'
require_relative './parens_v1'

describe 'parens(n)' do
  context 'when the input is 0' do
    it 'should return an array with 1 empty string' do
      expect(parens(0)).to eq([''])
    end
  end

  context 'when the input is 1' do
    it 'should return 1 set of balanced parenthesis' do
      expect(parens(1)).to eq(['()'])
    end
  end

  context 'when the input is 2' do
    it 'should return 2 set of balanced parenthesis' do
      expect(parens(2)).to eq(['(())', '()()'])
    end
  end

  context 'when the input is 3' do
    it 'should return 3 set of balanced parenthesis' do
      expect(
        parens(3)
      ).to eq(['((()))', '(()())', '(())()', '()(())', '()()()'])
    end
  end
end

File: algds/codewars/4kyu/all-balanced-parenthesis/js/parens.spec.js
import { parens } from './parens_v1.js';

describe('parens(n)', () => {
  describe('when the input is 0', () => {
    it('should return an array with 1 empty string', () => {
      expect(parens(0)).toEqual([''])
    });
  });

  describe('when the input is 1', () => {
    it('should return 1 set of balanced parenthesis', () => {
      expect(parens(1)).toEqual(['()'])
    });
  });

  describe('when the input is 2', () => {
    it('should return 2 set of balanced parenthesis', () => {
      expect(parens(2)).toEqual(['(())', '()()'])
    });
  });

  describe('when the input is 3', () => {
    it('should return the five possible ', () => {
      expect(
        parens(3)
      ).toEqual(['((()))', '(()())', '(())()', '()(())', '()()()'])
    });
  });
});
